File: python/704-Binary-Search.py
class Solution:
    # Space O(1)
    # Time O(logn)
    def search(self, nums: List[int], target: int) -> int:
        i, j = 0, len(nums) - 1

        while (i <= j):
            mid = math.floor(j + (i - j)/ 2)
            if nums[mid] == target:
                return mid
            if nums[mid] < target:
                i = mid + 1
            else:
                j = mid - 1

        return -1

File: c#/704-Binary-Search.cs
// Time O(log(n))
// Space O(1)
public class Solution
{
    public int Search(int[] nums, int target)
    {
        int l = 0,
            r = nums.Length - 1;

        while (l <= r)
        {
            int mid = l + (r - l) / 2;
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] > target)
                r = mid - 1;
            else
                l = mid + 1;
        }
        return -1;
    }
}
